FeedAPIChallenge/Feed API/RemoteFeedLoader.swift

```diff
@@ -19,6 +30,30 @@ public final class RemoteFeedLoader: FeedLoader {
 	}
 
 	public func load(completion: @escaping (FeedLoader.Result) -> Void) {
-		fatalError("Must be implemented")
+		client.get(from: self.url) { [weak self] result in
+			guard let _ = self else { return }
+
+			guard let (data, httpResponse) = try? result.get() else {
+				return completion(.failure(Error.connectivity))
+			}
+
+			guard httpResponse.statusCode == 200 else {
+				return completion(.failure(Error.invalidData))
+			}
+
+			let decoder = JSONDecoder()
+			decoder.keyDecodingStrategy = .convertFromSnakeCase
+			guard let response = try? decoder.decode(FeedImageResponse.self, from: data) else {
+				return completion(.failure(Error.invalidData))
+			}
```

Comment:
Since both of these statements complete with the same error, consider joining them into a single guard statement:
```suggestion
			let decoder = JSONDecoder()
			decoder.keyDecodingStrategy = .convertFromSnakeCase
			guard httpResponse.statusCode == 200,
			      let response = try? decoder.decode(FeedImageResponse.self, from: data) else {
				return completion(.failure(Error.invalidData))
			}
```

---

If you decide against using `keyDecodingStrategy` (following from my previous comment suggesting `CodingKeys` instead), it's possible to inline the `JSONDecoder` initialization:
```suggestion
			guard httpResponse.statusCode == 200,
			      let response = try? JSONDecoder().decode(FeedImageResponse.self, from: data) else {
				return completion(.failure(Error.invalidData))
			}
```